src/main/java/org/apache/maven/plugins/dependency/tree/JsonDependencyNodeVisitor.java

```diff
@@ -56,14 +58,20 @@ public boolean visit(DependencyNode node) {
      * @param node  the node to write
      */
     private void writeRootNode(DependencyNode node) {
-        Set<DependencyNode> visited = new HashSet<>();
-        int indent = 2;
-        StringBuilder sb = new StringBuilder();
-        sb.append("{").append("\n");
-        writeNode(indent, node, sb, visited);
-        sb.append("}").append("\n");
-        writer.write(sb.toString());
+        try {
+            StringBuilder sb = new StringBuilder();
+            Set<DependencyNode> visited = new HashSet<>();
+            int indent = 2;
+            sb.append("{").append("\n");
+            writeNode(indent, node, sb, visited);
+            sb.append("}").append("\n");
+            writer.write(sb.toString());
+            writer.flush();
+        } catch (IOException e) {
+            throw new UncheckedIOException("Failed to write JSON output", e);
```

Comment:
not needed; this method can and should throw the IOException directly since it's private